### ISSUE TITLE
Illuminate - Schema creator

### ISSUE THREAD

**Issue by lukasmahl** [themosis/framework#499]
Hey there!

I am new to themosis and I am currently developing my first Wordpress plugin with it. So far so good - I'm currently creating some models and I was asking myself if I am amble to use the Schema::create command as described in the Laravel documentation for creating the corresponding MySQL tables: https://laravel.com/docs/5.6/migrations with Themosis.

I was trying to use it but I do get a 'Call to a member function connection() on null' error when calling protected static function getFacadeAccessor(). 

Just wanted to get a quick feedback if this is even possible with the current version.

Thanks a lot!

**Comment by jlambe** [comment]
It is not available as our default service provider only registers the `capsule` instance. Other utilities as migration tools are not directly available but I'm pretty sure you can add them yourself by providing your own service provider.

**Comment by lukasmahl** [comment]
Thank you for your quick response! 

I found this Laravel Service Provider, which registers the $app['db'] instance here: https://github.com/illuminate/database/blob/master/DatabaseServiceProvider.php - so basically I do need to add the code in here as my own service provider?

Thank you very much for your help - much appreciated.

**Comment by jlambe** [comment]
Basically yes. Just make sure to grab the same `CapsuleManager` instance used by the framework and not recreating a separate one.

You can grab the instance from a global like so: `$capsule = $GLOBALS['themosis.capsule'];`

**Comment by lukasmahl** [comment]
Okay! The class itself does not inherit from CapsuleManager nor does it create any capsule instance, unsure where to add it then. And when I add the service (which is working properly) I still get the same error:

Call to a member function connection() on null
`vendor/illuminate/support/Facades/Schema.php
{
    /**
     * Get a schema builder instance for a connection.
     *
     * @param  string  $name
     * @return \Illuminate\Database\Schema\Builder
     */
    public static function connection($name)
    {
        return static::$app['db']->connection($name)->getSchemaBuilder();
    }
 
    /**
     * Get a schema builder instance for the default connection.
     *
     * @return \Illuminate\Database\Schema\Builder
     */
    protected static function getFacadeAccessor()
    {
        **return static::$app['db']->connection()->getSchemaBuilder();**
    }
}
 `

By the way, if you think that this will be too much effort explaining, then let me know - then I'll just use $wpdb.

Thanks!

**Comment by jlambe** [comment]
Not sure how long it takes to find the solution. One thing to check is the alias of `db` from the service container. You need to find out which object/class it is expecting.

Now you get a `null`, which basically means that nothing has been instantiated into the service container. It might be called to early in the stack as well.

**Comment by lukasmahl** [comment]
I managed it! I am 100% sure that this is not the most beautiful and probably the wrong way to do it, but it's working fine! 

I modified this code in the Schema facade class in the Laravel core:
`return static::$app['db']->connection()->getSchemaBuilder();`

Changed it to match with the Capsule default connection:
`global $capsule;
return $capsule->getConnection()->getSchemaBuilder();`

The Schema facade class is expecting a database connection object, I guess if doing it correct I'd have to inject that with a service container.

Thank you for your help! I'm happy with the current solution for the moment 👍
